tests/cloner_test.go

```diff
@@ -0,0 +1,133 @@
+			nil))
+		Expect(err).ToNot(HaveOccurred())
+		err = f.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, targetPvc.Name)
+		Expect(err).ToNot(HaveOccurred())
+
+		sourcePod, err := f.FindPodByPrefix(common.CLONER_SOURCE_PODNAME)
+		Expect(err).ToNot(HaveOccurred())
+		targetPod, err := f.FindPodByPrefix(common.CLONER_TARGET_PODNAME)
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(strings.Compare(sourcePod.Spec.NodeName, targetPod.Spec.NodeName) == 0).To(BeTrue())
+
+		err = f.WaitTimeoutForPodStatus(sourcePod.Name, v1.PodSucceeded, cloneCompleteTimeout)
+		Expect(err).ToNot(HaveOccurred())
+		err = f.WaitTimeoutForPodStatus(targetPod.Name, v1.PodSucceeded, cloneCompleteTimeout)
+		Expect(err).ToNot(HaveOccurred())
+		// Clone is completed, verify the content matches the source.
+		Expect(verifyTargetContent(f, f.Namespace, targetPvc)).To(BeTrue())
+		// Clean up PVC, the AfterEach will also clean it up, through the Namespace delete.
+		if targetPvc != nil {
+			err = f.DeletePVC(targetPvc)
+			Expect(err).ToNot(HaveOccurred())
+		}
+	})
+
+	It("Cloner should clone data from PVC in name space to other PVC in different name space", func() {
+		targetNs, err := f.CreateNamespace(f.NsPrefix, map[string]string{
+			framework.NsPrefixLabel: f.NsPrefix,
+		})
+		Expect(err).NotTo(HaveOccurred())
+		f.AddNamespaceToDelete(targetNs)
+		// Create targetPvc in new NS.
+		targetPvc, err := utils.CreatePVCFromDefinition(f.K8sClient, targetNs.Name, utils.NewPVCDefinition(
+			"target-pvc",
+			"1G",
+			map[string]string{controller.AnnCloneRequest: f.Namespace.Name + "/" + sourcePVCName},
+			nil))
+		Expect(err).ToNot(HaveOccurred())
+		err = utils.WaitForPersistentVolumeClaimPhase(f.K8sClient, targetNs.Name, v1.ClaimBound, targetPvc.Name)
+
+		sourcePod, err := f.FindPodByPrefix(common.CLONER_SOURCE_PODNAME)
+		Expect(err).ToNot(HaveOccurred())
+		targetPod, err := utils.FindPodByPrefix(f.K8sClient, targetNs.Name, common.CLONER_TARGET_PODNAME, common.CDI_LABEL_SELECTOR)
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(strings.Compare(sourcePod.Spec.NodeName, targetPod.Spec.NodeName) == 0).To(BeTrue())
+
+		err = f.WaitTimeoutForPodStatus(sourcePod.Name, v1.PodSucceeded, cloneCompleteTimeout)
+		Expect(err).ToNot(HaveOccurred())
+		err = utils.WaitTimeoutForPodStatus(f.K8sClient, targetPod.Name, targetNs.Name, v1.PodSucceeded, cloneCompleteTimeout)
+		Expect(err).ToNot(HaveOccurred())
+		// Clone is completed, verify the content matches the source.
+		Expect(verifyTargetContent(f, targetNs, targetPvc)).To(BeTrue())
+		// Clean up PVC, the AfterEach will also clean it up, through the Namespace delete.
+		if targetPvc != nil {
+			err = utils.DeletePVC(f.K8sClient, targetNs.Name, targetPvc)
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+	})
+
+})
+
+func verifyTargetContent(f *framework.Framework, namespace *v1.Namespace, pvc *v1.PersistentVolumeClaim) bool {
+	frameworkNs := f.Namespace
+	f.Namespace = namespace
```

Comment:
I don't like having to save the ns in the Framework so that it can be re-assigned just so the `CreateExecutorPodWithPVC` method has a ns to reference. In these cases I'd rather see the ns as a separate arg to `CreateExecutorPodWithPVC`. What do you think?

Reply:
I don't like it either, the problem is the exec_util.go is tied to the framework with no ability to pass in a namespace. So I had 2 options.
1. Temporarily override the NS in the framework, which is ugly
2. Rewrite the exec_util to be split into two piece, one utils and one framework, like the rest.

I opted for 1 for speed as I can't see many places besides the cloner needing more than one namespace.

Reply:
Is there a way to utilize `Framework.Namespace2` here?

Reply:
The problem is in f.ExecShellInPod, that is defined in exec_util, I don't have a non framework version of the function. So I am forced to either completely rewrite exec_util, or mess with the namespace in framework like I did.

Reply:
I can actually make a fairly small modification to exec_util to accept a namespace.

Reply:
Alex, yes I think it's best (for future e2e/functional test authors) to do this a cleaner way. It would be a shame, imo, to see this pattern copied across several tests. In my view, the time to change _ exec_util.go_ would be well spent if it saves you from re-assigning to the framework's primary ns just to be able to call an exec func.